indra/newview/llviewerobject.cpp

```diff
@@ -1496,20 +1417,45 @@ U32 LLViewerObject::processUpdateMessage(LLMessageSystem *mesgsys,
 				if (mData)
 				{
 					delete [] mData;
-				}
+                    mData = NULL;
+                }
 
-				// Check for appended generic data
-				S32 data_size = mesgsys->getSizeFast(_PREHASH_ObjectData, block_num, _PREHASH_Data);
-				if (data_size <= 0)
-				{
-					mData = NULL;
-				}
-				else
-				{
-					// ...has generic data
-					mData = new U8[data_size];
```

Comment:
Older viewers with this code will end up allocating 1 byte for every linkset root object and never using the new data.   This has the potential to impact the viewer - we need to measure this and perhaps make it smarter to not send to old viewers.